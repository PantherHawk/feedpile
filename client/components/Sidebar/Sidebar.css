.sidebar-container {
  position: fixed;
  width: 250px;
  float: left;
  height: 100vh;
  border-right: 1px solid black;
  background-color: #f2f2f2;
}

.sidebar-logo {
  height: 80px;
  width: auto;
  display:block;
  margin:auto;
  padding-top: 50px;
}

.logo-name {
  text-align: center;
  margin-top: 10px;
  margin-bottom: 40px;
}
