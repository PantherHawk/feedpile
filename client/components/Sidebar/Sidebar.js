import React from 'react';
// import SidebarAll from '../SidebarAll/SidebarAll';
import {List, ListItem} from 'material-ui/List';
import Inbox from 'material-ui/svg-icons/content/inbox/';
// import ContentInbox from 'material-ui-icons/';
import './Sidebar.css';
import { connect } from 'react-redux';
import { toggleListItem } from '../../actions'

const Sidebar = (props) => {
  return (
    <div className="sidebar-container">
      <img className="sidebar-logo" src={"https://s3-us-west-1.amazonaws.com/thesis.feedpile.images/feedpile.png"} />
      <h2 className="logo-name">Feedpile</h2>
      {/* <SidebarAll /> */}
      <List>
        <ListItem primaryText="Sent mail" leftIcon={<Inbox />} />
        <ListItem primaryText="Drafts" leftIcon={<Inbox />} />
        <ListItem
          primaryText="Inbox"
          leftIcon={<Inbox />}
          initiallyOpen={true}
          primaryTogglesNestedList={true}
          nestedItems={[
            <ListItem
              key={1}
              primaryText="Starred"
              leftIcon={<Inbox />}
            />,
            <ListItem
              key={2}
              primaryText="Sent Mail"
              leftIcon={<Inbox />}
              disabled={true}
              nestedItems={[
                <ListItem key={1} primaryText="Drafts" leftIcon={<Inbox />} />,
              ]}
            />,
            <ListItem
              key={3}
              primaryText="Inbox"
              leftIcon={<Inbox />}
              open={props.open}
              onNestedListToggle={props.dispatchToggle}
              nestedItems={[
                <ListItem key={1} primaryText="Drafts" leftIcon={<Inbox />} />,
              ]}
            />,
          ]}
        />
      </List>
    </div>
  )
}

const mapStateToProps = (state) => {
  return {
    open: state.open
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    dispatchToggle(item) {
      dispatch(toggleListItem(item))
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Sidebar);
// export default Sidebar;
