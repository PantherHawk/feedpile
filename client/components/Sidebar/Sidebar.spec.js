import React from 'react';
import Sidebar from './Sidebar';
// import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json'

test('Sidebar snapshot test', () => {
  // const component = renderer.create(<Sidebar />);
  const component = shallow(<Sidebar />);
  // const tree = component.toJSON();
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});
