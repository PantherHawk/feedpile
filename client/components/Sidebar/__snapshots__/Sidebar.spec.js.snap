// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar snapshot test 1`] = `
<div
  className="sidebar-container"
>
  <img
    className="sidebar-logo"
    src="test-file-stub"
  />
  <h2
    className="logo-name"
  >
    FeedPile
  </h2>
  <SelectableList
    defaultValue={1}
  >
    <ListItem
      autoGenerateNestedIndicator={true}
      disableKeyboardFocus={false}
      disabled={false}
      initiallyOpen={false}
      insetChildren={false}
      leftIcon={<pure(List) />}
      nestedItems={Array []}
      nestedLevel={0}
      onClick={[Function]}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onNestedListToggle={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      open={null}
      primaryText="All Articles"
      primaryTogglesNestedList={false}
      secondaryTextLines={1}
      value={1}
    />
    <ListItem
      autoGenerateNestedIndicator={true}
      disableKeyboardFocus={false}
      disabled={false}
      initiallyOpen={false}
      insetChildren={false}
      leftIcon={<pure(Star) />}
      nestedItems={Array []}
      nestedLevel={0}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onNestedListToggle={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      open={null}
      primaryText="Starred"
      primaryTogglesNestedList={false}
      secondaryTextLines={1}
      value={2}
    />
    <ListItem
      autoGenerateNestedIndicator={true}
      disableKeyboardFocus={false}
      disabled={false}
      initiallyOpen={false}
      insetChildren={false}
      leftIcon={<pure(Archive) />}
      nestedItems={Array []}
      nestedLevel={0}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onNestedListToggle={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      open={null}
      primaryText="Archived"
      primaryTogglesNestedList={false}
      secondaryTextLines={1}
      value={3}
    />
  </SelectableList>
  <span
    className="refresh-icon"
  >
    <IconButton
      className="refresh-icon"
      disableTouchRipple={false}
      disabled={false}
      iconStyle={
        Object {
          "height": 36,
          "width": 36,
        }
      }
      style={
        Object {
          "height": 72,
          "padding": 16,
          "width": 72,
        }
      }
      tooltipPosition="bottom-center"
      touch={false}
    >
      <pure(Refresh) />
    </IconButton>
  </span>
  <span
    className="archive-icon"
  >
    <IconButton
      className="add-icon"
      disableTouchRipple={false}
      disabled={false}
      iconStyle={
        Object {
          "height": 36,
          "width": 36,
        }
      }
      onClick={[Function]}
      style={
        Object {
          "height": 72,
          "padding": 16,
          "width": 72,
        }
      }
      tooltipPosition="bottom-center"
      touch={false}
    >
      <pure(AddCircle) />
    </IconButton>
  </span>
  <span
    className="archive-icon"
  >
    <IconButton
      className="archive-icon"
      disableTouchRipple={false}
      disabled={false}
      iconStyle={
        Object {
          "height": 36,
          "width": 36,
        }
      }
      style={
        Object {
          "height": 72,
          "padding": 16,
          "width": 72,
        }
      }
      tooltipPosition="bottom-center"
      touch={false}
    >
      <pure(Archive) />
    </IconButton>
  </span>
  <Connect(AddFeed) />
</div>
`;
